confluent_kafka_helpers/producer.py

```diff
@@ -107,15 +112,44 @@ def produce(self, value, key=None, topic=None, headers=None, **kwargs):
             raise TopicNotRegistered(f"Topic {topic} is not registered")
 
         if self.value_serializer:
-            value = self.value_serializer(value)
+            with tracer.start_span(name='kafka.serialize_message') as span:
+                span.set_attribute(
+                    attrs.MESSAGING_OPERATION_TYPE, attrs.MESSAGING_OPERATION_TYPE_VALUE_CREATE
+                )
+                value = self.value_serializer(value)
+
+        message_class = value.get("class") if isinstance(value, dict) else None
```

Comment:
Same here, introspect the message and use the class name as part of the resource name